tests/test_bincopy.py

```diff
@@ -569,40 +569,69 @@ def test_performance(self):
         binfile = bincopy.BinFile()
         binfile.add_srec(srec)
 
-    def test_command_line_info_help(self):
-        argv = ['bincopy', 'info', '--help']
-        output = """usage: bincopy info [-h] binfile [binfile ...]
+    def test_command_line_help(self):
+        commands_descriptions=[
+            ('info','Print general information about given file(s).'),
+            ('as_hexdump','Print hexdump to stdout.'),
+            ('as_srec','Print srec to stdout.'),
+            ('as_ihex','Print ihex to stdout.')]
 
-Print general information about given file(s).
+        for tupl in commands_descriptions:
+            argv = ['bincopy', tupl[0], '--help']
+            output = """usage: bincopy {} [-h] binfile [binfile ...]
+
+{}
 
 positional arguments:
   binfile     One or more binary format files.
 
 optional arguments:
   -h, --help  show this help message and exit
-"""
+""".format(tupl[0],tupl[1])
+
+            with self.assertRaises(SystemExit) as cm:
+                self._test_command_line_raises(argv, output)
+
+            self.assertEqual(cm.exception.code, 0)
+
+
+    def test_command_line_non_existing_file(self):
+        commands=['info', 'as_hexdump', 'as_srec', 'as_ihex']
+        for command in commands:
+            argv = ['bincopy', command, 'non-existing-file']
+            output = ""
+
+            with self.assertRaises(SystemExit) as cm:
+                self._test_command_line_raises(argv, output)
+
+            self.assertEqual(cm.exception.code,
+                            "[Errno 2] No such file or directory: 'non-existing-file'")
 
-        with self.assertRaises(SystemExit) as cm:
-            self._test_command_line_raises(argv, output)
+    def test_command_line_non_existing_file_debug(self):
+        commands=['info', 'as_hexdump', 'as_srec', 'as_ihex']
+        for command in commands:
+            argv = ['bincopy', '--debug', command, 'non-existing-file']
+            output = ""
 
-        self.assertEqual(cm.exception.code, 0)
+            with self.assertRaises(IOError):
+                self._test_command_line_raises(argv, output)
 
-    def test_command_line_info_non_existing_file(self):
-        argv = ['bincopy', 'info', 'non-existing-file']
-        output = ""
 
-        with self.assertRaises(SystemExit) as cm:
-            self._test_command_line_raises(argv, output)
+    def test_command_line_dump_commands_one_file(self):
+        test_file="tests/files/empty_main.s19"
+        binfile = bincopy.BinFile(test_file)
 
-        self.assertEqual(cm.exception.code,
-                         "[Errno 2] No such file or directory: 'non-existing-file'")
+        command_func_pairs=[
+            ('as_hexdump',binfile.as_hexdump),
```

Comment:
Change to `('as_hexdump', binfile.as_hexdump())`, and remove the function call in the loop.